src/main/java/jpos/config/simple/xml/JavaxRegPopulator.java

```diff
@@ -0,0 +1,450 @@
+import jpos.util.JposEntryUtility;
+import jpos.util.tracing.Tracer;
+import jpos.util.tracing.TracerFactory;
+import org.w3c.dom.*;
+import org.xml.sax.*;
+import org.xml.sax.ext.DefaultHandler2;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import java.io.*;
+import java.net.URL;
+import java.text.DateFormat;
+import java.util.*;
+
+/**
+ * Simple implementation of the JposRegPopulator that loads and saves
+ * the entries in XML using Javax API.
+ * NOTE: this class must define a public no-argument constructor so that it may be
+ * created via reflection when it is defined in the jpos.properties as
+ * the jpos.config.regPopulatorClass
+ * @see jpos.util.JposProperties#JPOS_REG_POPULATOR_CLASS_PROP_NAME
+ * @since 4.0
+ * @author M. Conrad (martin.conrad@freenet.de)
+ */
+public class JavaxRegPopulator
+        extends AbstractRegPopulator
+        implements XmlRegPopulator {
+    /**
+     * Default ctor
+     */
+    public JavaxRegPopulator() {
+        super(JavaxRegPopulator.class.getName());
+    }
+
+    /**
+     * 1-arg constructor that takes the unique ID
+     *
+     * @param s the unique ID string
+     * @since 1.3
+     */
+    public JavaxRegPopulator(String s) {
+        super(s);
+    }
+
+    //-------------------------------------------------------------------------
+    // Public methods
+    //
+
+    @Override
+    public String getClassName() {
+        return JavaxRegPopulator.class.getName();
+    }
+
+    @Override
+    public URL getEntriesURL() {
+        URL url = null;
+
+        if (getPopulatorFileURL() != null &&
+                !getPopulatorFileURL().equals(""))
+            try {
+                url = new URL(getPopulatorFileURL());
+            } catch (Exception e) {
+                tracer.println("getEntriesURL: Exception.message=" +
+                        e.getMessage());
+            }
+        else
+            url = createURLFromFile(new File(getPopulatorFileName()));
+
+        tracer.println("getPopulatorFileURL()=" + getPopulatorFileURL());
+        tracer.println("getPopulatorFileName()=" + getPopulatorFileName());
+
+        return url;
+    }
+
+
+    @Override
+    public void save(Enumeration entries)
+            throws Exception {
+        if (isPopulatorFileDefined())
+            save(entries, getPopulatorFileOS());
+        else
+            try (FileOutputStream os = new FileOutputStream(DEFAULT_XML_FILE_NAME)) {
+                save(entries, os);
+            }
+    }
+
+    @Override
+    public void save(Enumeration entries, String fileName)
```

Comment:
It's good to get rid of this compiler warning at all, as we will not solve it by intention...

```suggestion
    @SuppressWarnings("unchecked")
    @Override
    public void save(Enumeration entries, String fileName)
```